components/gitpod-protocol/src/util/tracing.ts

```diff
@@ -138,13 +138,21 @@ export namespace TraceContext {
         }
         const namespace = _namespace ? `${_namespace}.` : '';
 
-        for (const k of Object.keys(keyValueMap)) {
-            const v = keyValueMap[k];
-            if (typeof v === 'object') {
-                addNestedTags(ctx, v, `${namespace}${k}`);
-            } else {
-                ctx.span.setTag(`${namespace}${k}`, v);
+        try {
+            for (const k of Object.keys(keyValueMap)) {
+                const v = keyValueMap[k];
+                // oh my JavaScript... `typeof null` and `typeof undefined` return `object`
```

Comment:
nit: not exactly, but ok. (typeof undedinfed === 'undefined')

Reply:
http://wiki.ecmascript.org/doku.php?id=harmony:typeof_null

Reply:
Took over Jan's proposal: https://github.com/gitpod-io/gitpod/pull/7118#pullrequestreview-826303507